homeassistant/components/switch/wemo.py

```diff
@@ -199,9 +201,33 @@ def turn_off(self, **kwargs):
         self.wemo.off()
         self.schedule_update_ha_state()
 
-    def update(self):
-        """Update WeMo state."""
-        self._update(force_update=True)
+    async def async_added_to_hass(self):
+        """Wemo switch added to HASS."""
+        # Define inside async context so we know our event loop
+        self._update_lock = asyncio.Lock()
+
+    async def async_update(self):
+        """Update WeMo state.
+
+        Wemo has an aggressive retry logic that sometimes can take over a
+        minute to return. If we don't get a state after 5 seconds, assume the
+        Wemo switch is unreachable. If update goes through, it will be made
+        available again.
+        """
+        if self._update_lock.locked():
```

Comment:
This doesn't protect `_update` from being called excessively both from callback and poll.